src/theories/Category.jl

```diff
@@ -197,3 +199,34 @@ abstract type TightExpr{T} <: GATExpr{T} end
   compose(f::Hom, g::Hom) = associate_unit(new(f,g; strict=true), id)
   transitive(t::Tight, u::Tight) = associate_unit(new(t,u; strict=true), reflexive)
 end
+
+"""
+Theory of a pointed category.
+We axiomatize a category equipped with zero morphisms 
+and a distinguished weak zero object. That is, we don't axiomatize 
+that morphisms to and from the zero object `Z` are unique,
+although they will be in an instance freely generated by a category.
+
+A functor from an ordinary category into a freely pointed category, 
+equivalently, a pointed category in which no two nonzero maps 
+compose to a zero map, is a good notion
+of partially-defined functor into the complement of the
+zero maps and zero object in the codomain.
+"""
+@theory ThPtCategory{Ob,Hom,ZeroHom} <: ThCategory{Ob,Hom} begin
+  Z()::Ob
+  ZeroHom(dom::Ob,codom::Ob)::TYPE
+  z(A::Ob,B::Ob)::ZeroHom(A,B)
+  hom(x::ZeroHom(A,B))::Hom(A,B)⊣(A::Ob,B::Ob)
+  compose(x::ZeroHom(A,B),f::(B→C))::ZeroHom(A,C)⊣(A::Ob,B::Ob,C::Ob)
+  compose(g::(A→B),x::ZeroHom(B,C))::ZeroHom(A,C)⊣(A::Ob,B::Ob,C::Ob)
+  z₁==z₂⊣(A::Ob,B::Ob,z₁::ZeroHom(A,B),z₂::ZeroHom(A,B))
```

Comment:
I think we can get rid of `ZeroHom`, have `zero(A::Ob, B::Ob)::Hom(A,B)`, and equation

```julia
compose(zero(A,B), f) == zero(A,C) ⊣ (A::Ob, B::Ob, C::Ob, f::Hom(B,C))
```

plus other equation. Will give `HomExpr{:zero}` type in syntax.